homeassistant/components/device_tracker/icloud.py

```diff
@@ -47,24 +47,22 @@ def keep_alive(now):
         api.authenticate()
         _LOGGER.info("Authenticate against iCloud")
 
-    track_utc_time_change(hass, keep_alive, second=0)
+    track_utc_time_change(hass, keep_alive, second=0, minute=4)
```

Comment:
This will only track the time change when the minute == 4. Which is once an hour. Instead, make it `/4`. 

Did you test these changes?